### PR TITLE
gh-105844: Use devguide terminology to denote versions

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-105844

----
:books: Documentation preview :books:: https://cpython-previews--105882.org.readthedocs.build/

### CODE REVIEW

File: Doc/install/index.rst [hunk 1]
```diff
@@ -696,7 +696,7 @@ is supplied to suppress this behaviour.  So you could simply edit
    import sys
    sys.path.append('/www/python/')
 
-However, if you reinstall the same minor version of Python (perhaps when
+However, if you reinstall the same feature version of Python (perhaps when
```
Comment (erlend-aasland) [inline Doc/install/index.rst]:
```suggestion
However, if you reinstall the same feature release of Python (perhaps when
```

Comment (AlexWaygood) [reply Doc/install/index.rst]: I might actually keep this one as "minor version" -- while I agree that "feature release/bugfix release" is a clearer terminology in general, in this specific case both "feature release" and "feature version" read more awkwardly than "minor version", in my opinion. But I'm curious for @CAM-Gerlach's thoughts.

**Review by AlexWaygood — APPROVED** [review]

File: Doc/install/index.rst (hunk 1 above)
Comment (erlend-aasland) [reply Doc/install/index.rst]: Yeah, I pondered about this myself. I'm not sure what reads best here. Perhaps we can phrase the sentence in a better way.

Comment (erlend-aasland) [reply Doc/install/index.rst]: "Reinstall" also feels weird here. I do not think of upgrading to a new patch/micro version as reinstalling.

**Comment by AlexWaygood** [comment]
Not sure what's up with the doctest -- looks like all the individual doctests passed but Python segfaulted somewhere...?

File: Doc/install/index.rst (hunk 1 above)
Comment (CAM-Gerlach) [reply Doc/install/index.rst]: These docs are super-outdated, are the ancient legacy version of the newer `installing.rst` doc, and are scheduled for removal in 3.12 anyway as they are distutils-specific, which is being removed in 3.12 (I've just been waiting on confirmation of Setuptools plans/needs here to actually do the removal). I would definitely not bother touching them now (or spending more time thinking about this, heh).

```suggestion
However, if you reinstall the same minor version of Python (perhaps when
```

**Review by CAM-Gerlach — COMMENTED** [review]
One comment, otherwise LGTM, thanks!

File: Doc/install/index.rst (hunk 1 above)
Comment (erlend-aasland) [reply Doc/install/index.rst]: SGTM

**Review by CAM-Gerlach — APPROVED** [review]
LGTM, thanks @erlend-aasland !

**Comment by erlend-aasland** [comment]
Thanks for the reviews!

**Pull request merged**